src/game/ingame_menu.c

```diff
@@ -256,298 +385,341 @@ static u8 *alloc_ia8_text_from_i1(u16 *in, s16 width, s16 height) {
-                        rgbaColors[(8 - (colorLoop - strPos)) / 2] += ((str[strPos] - diffTmp) & 0x0F);
-                    }
-                }
-                strPos--;
-                if (customColor == 1) {
-                    gDPSetEnvColor(gDisplayListHead++, rgbaColors[0],
-                                                    rgbaColors[1],
-                                                    rgbaColors[2],
-                                                    rgbaColors[3]);
-                } else if (customColor == 2) {
-                    gDPSetEnvColor(gDisplayListHead++, 255, 255, 255, 255); // TODO: Is it possible to retrieve the original color that was set before print_generic_string was called?
-                    customColor = 0;
-                }
-                break;
-            case DIALOG_CHAR_DAKUTEN:
-                mark = DIALOG_MARK_DAKUTEN;
-                break;
-            case DIALOG_CHAR_PERIOD_OR_HANDAKUTEN:
-                mark = DIALOG_MARK_HANDAKUTEN;
-                break;
-            case DIALOG_CHAR_NEWLINE:
+            // Newline
+            case '\n':
                 gSPPopMatrix(gDisplayListHead++, G_MTX_MODELVIEW);
-                create_dl_translation_matrix(MENU_MTX_PUSH, x, y - (lineNum * MAX_STRING_WIDTH), 0.0f);
+                if (lineNum == maxLines) {
+                    return strPos + 1;
+                }
+                create_dl_translation_matrix(MENU_MTX_PUSH, x, y - (lineNum * sGenericFontLineHeight), 0.0f);
                 lineNum++;
+                // Can skip any queued spaces
+                queuedSpaces = 0;
                 break;
-            case DIALOG_CHAR_PERIOD:
-                create_dl_translation_matrix(MENU_MTX_PUSH, -2.0f, -5.0f, 0.0f);
-                render_generic_char(DIALOG_CHAR_PERIOD_OR_HANDAKUTEN);
-                gSPPopMatrix(gDisplayListHead++, G_MTX_MODELVIEW);
-                break;
-            case DIALOG_CHAR_SLASH:
-                create_dl_translation_matrix(MENU_MTX_NOPUSH, (f32)(gDialogCharWidths[DIALOG_CHAR_SPACE] * 2), 0.0f, 0.0f);
-                break;
-            case DIALOG_CHAR_MULTI_THE:
-                render_multi_text_string(STRING_THE);
-                break;
-            case DIALOG_CHAR_MULTI_YOU:
-                render_multi_text_string(STRING_YOU);
+
+            // Space
+            case ' ':
+                queuedSpaces++;
                 break;
-            case DIALOG_CHAR_SPACE:
-                // create_dl_translation_matrix(MENU_MTX_NOPUSH, (f32)(gDialogCharWidths[DIALOG_CHAR_SPACE]), 0.0f, 0.0f);
-                create_dl_translation_matrix(MENU_MTX_NOPUSH, CHAR_WIDTH_SPACE, 0.0f, 0.0f);
+
+            // Backslash / escape character: Force the following character to print normally.
+            // Note that you will have to type '\\' to use this so that the compiler doesn't
+            // interpret it as a real escape character. To render one backslash, use '\\\\'.
```

Comment:
because if you write "\\" in C the string itself will only contain one backslash. nothing i can do about that

Reply:
lmfao even github does it, i meant "\\\\" ofc

Reply:
I don't think a condition for backslash is necessary at all to be honest. Escaped characters are preprocessed; what's the point of double processing them here?

Reply:
@thecozies I'll let you push the issue if you disagree with this too but I'm personally done fighting it I think since it's not the biggest issue in the world.

Reply:
since i couldnt respond here before ill repeat it: the point is to be able to escape the ingame string, the double backslash is required to get past the first layer. the compiler processing isnt enough to handle escaping the game processing a %d or a @FF0000